### PR TITLE
Adding Board b-g431b-esc1 stm32 discovery kit

### PR DESCRIPTION
**Pull request by owennewo**
**Summary**

* Add support for *b-g431b-esc1* discovery kit

https://www.st.com/en/evaluation-tools/b-g431b-esc1.html

This board is one of the stm32 discovery kit and can be used to control bldc motors e.g. for drones.  It has a `stm32g431cbu6` (48pins) and uses a chip fairly similar to the existing `nucleo_g431rb` board (64pin).

This board does not have any arduino r3 pins but you can solder approximately 12 contacts (rx, tx, hall sensors, swdio, can, pwm).

**Validation**
I've uploaded a blink sketch to this board using this config (using platformio).

### CODE REVIEW

**Comment by fpistm** [comment]
Thanks @owennewo 
Will be added later when #1091 will be done

**Comment by owennewo** [comment]
FYI: Here is a matching PR for the board.json file in `platform-ststm32`:
https://github.com/platformio/platform-ststm32/pull/447

File: variants/DISCO_B_G431B_ESC1/PeripheralPins.c [hunk 1]
```diff
@@ -150,7 +150,7 @@ WEAK const PinMap PinMap_PWM[] = {
   //  {PB_9,  TIM17,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM17, 1, 0)}, // TIM17_CH1
   //  {PB_10, TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 3, 0)}, // TIM2_CH3
   //  {PB_11, TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 4, 0)}, // TIM2_CH4
-  {PB_13, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_TIM1, 1, 1)}, // TIM1_CH1N
+  //  {PB_13, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_TIM1, 1, 1)}, // TIM1_CH1N
```
Comment (fpistm) [inline variants/DISCO_B_G431B_ESC1/PeripheralPins.c:153]: Why this pin is incorrect ?
image

Comment (owennewo) [reply variants/DISCO_B_G431B_ESC1/PeripheralPins.c:153]: I think you are looking at a generic g431b pin usage - this board has been wired up with a specific purpose.  Page 15-16 of [b-g431b-esc1 user guide](https://www.st.com/resource/en/user_manual/dm00564746-electronic-speed-controller-discovery-kit-for-drones-with-stm32g431cb-stmicroelectronics.pdf
) explains why this change is correct.
pb13 is not connected on this discovery kit:
image

...and TIM1_CH1N is associated with PC13 on this board (it is connected to the negative drive gate of phase A motor driver)

image

File: boards.txt [hunk 2]
```diff
@@ -582,6 +582,19 @@ Disco.build.core=arduino
 Disco.build.board=Disco
 Disco.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial}
 
+# DISCO_B_G431B_ESC1 board
+Disco.menu.pnum.DISCO_B_G431B_ESC1=B-G431B-ESC1
+Disco.menu.pnum.DISCO_B_G431B_ESC1.node="NODE_G431CB,NOD_G431CB"
```
Comment (bg1729) [inline boards.txt:587]: I had to add "DIS_G431CB" to the node line above in order to get the programmer to recognise the storage device. I was using a brand new board (no changes to the ST Link FW).

**Comment by fpistm** [comment]
Hi @owennewo 
I've included your PR in #1091, it has been reformatted to the new variant definition way and made some fixes.
So I close this one as it is superseded by #1091.
Thanks again for your contribution.

**Pull request closed without merging**